tests/insert_reference/test.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+set -u
+cd "$(dirname "$0")"
+
+# Code used when the script ends
+error_code=0
+
+
+# Verify the differences between two files and 
+compare_file () {
+    output=$(diff -w $1 $2)
+    if [ $? -eq 0 ]; then
+        printf "Test $1 - OK\n"
+    else
+        printf "|-Error\n%s\n" "$output"
+        error_code=1
+    fi
+}
+
+# Execute distar command with an existing source and an existing target.
+# The target contains a portion of code from the source.
+../distar --track source.ml target.html > insert_reference.output 2>&1
```

Comment:
This looks strange: are you assuming that the distar executable is in `tests`? Are you assuming that this script must be executed from `tests`? In any case, these choices do not look very robust to me. I should be able to run the `test.sh` from anywhere.